### PR TITLE
derivesToLambda (java) and build script changes

### PR DESCRIPTION
**Pull request by joshdorsey**
Hey, I split some of the classes out into separate files to clean things up a bit.  There's a CFG class that basically just wraps the map you parsed the cfg to in convenience functions.  It seemed weird to have that class pull the map apart and store lots of instance variables for the sets and things so for now most of the getter/util functions just start from the map and get to whatever information they're trying to get.  That way if we update the map we don't need to worry about any other instance variables or anything.  There's all-lowercase functions that return streams of things, because large parts of the algorithms you can just build onto stream pipelines from those functions.  There's also accessor-style that collect the streams to lists/sets.

### CODE REVIEW

File: java/compilers/CFG.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        return terminals().collect(Collectors.toSet());
+    }
+
+    Set<Symbol> getNonTerminals() {
+        return nonterminals().collect(Collectors.toSet());
+    }
+    //</editor-fold>
+
+    //<editor-fold desc="Algorithms">
+    boolean derivesToLambda(Symbol s, Stack<Tuple<Rule, Symbol>> recurseStack) {
```
Comment (bobsmith947) [inline java/compilers/CFG.java:60]: We can use ArrayDeque rather than Stack. The latter is essentially deprecated in favor of the former.

**Review by bobsmith947 — COMMENTED** [review]
Good stuff, thanks! See my comment regarding a minor change.

File: java/compilers/CFG.java (hunk 1 above)
Comment (joshdorsey) [reply java/compilers/CFG.java:60]: Good call, I'll just make that change in master.

Comment (joshdorsey) [reply java/compilers/CFG.java:60]: Done: a710fb3e02

**Pull request merged**